server/src/test/java/org/elasticsearch/action/admin/cluster/migration/GetFeatureUpgradeStatusResponseTests.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.migration;
+
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.test.AbstractWireSerializingTestCase;
+
+import java.io.IOException;
+import java.util.List;
+
+public class GetFeatureUpgradeStatusResponseTests extends AbstractWireSerializingTestCase<GetFeatureUpgradeStatusResponse> {
+
+    @Override
+    protected Writeable.Reader<GetFeatureUpgradeStatusResponse> instanceReader() {
+        return GetFeatureUpgradeStatusResponse::new;
+    }
+
+    @Override
+    protected GetFeatureUpgradeStatusResponse createTestInstance() {
+        return new GetFeatureUpgradeStatusResponse(
+            List.of(),
```

Comment:
I think we should generate feature statuses here (at least sometimes), rather than just always using an empty list.